bot.js

```diff
@@ -1,55 +1,135 @@
-const URL_RANDOM = 'https://api.nasa.gov/planetary/apod?api_key=' + process.env.API_KEY + '&count=1';
 const bot = new Telegraf(process.env.BOT_TOKEN)
 
+const URL = process.env.URL_API + process.env.API_KEY;
+const URL_RANDOM = process.env.URL_API + process.env.API_KEY + '&count=1';
+
+const ONE_HOUR = 3600*10^3;
+
 //APOD - Astronomy picture of the day
 async function getJSONAPOD(url){
   try {
     return axios.get(url);
   } catch (e) {
     console.log(e);
+    throw 'Failed to retrieve JSON from ' + url;
   }
 };
 
+async function sendImageURL(ctx, json) {
+  if (json.data[0]) {
+    await ctx.replyWithPhoto({url:json.data[0].url}, {caption: json.data[0].title});
+    console.log("Random PhotoURL sent.");
+  }else {
+    await ctx.replyWithPhoto({url:json.data.url}, {caption: json.data.title});
+    console.log("PhotoURL sent.");
+  }
+};
+//Lançar erro nas funções send e tratar na main
+async function sendVideoURL(ctx, json) {
+  if (Array.isArray(json)) {
+    try {
+      await ctx.reply({url:json.data[0].url}, {caption: json.data[0].title});
+      await ctx.reply(json.data[0].explanation);
+      console.log("Random VideoURL sent.");
+    } catch (e) {
+      await ctx.reply('Falha ao recuperar a imagem do dia. :(');
+      console.log("Falhou.");
+      console.log(e);
+    }
+    await ctx.reply({url:json.data[0].url}, {caption: json.data[0].title});
+    await ctx.reply(json.data[0].explanation);
+    console.log("Random VideoURL sent.");
+  }else {
+    try {
+      await ctx.reply({url:json.data.url}, {caption: json.data.title});
+      await ctx.reply(json.data.explanation);
+      console.log("VideoURL sent.");
+    } catch (e) {
+      await ctx.reply('Falha ao recuperar a imagem do dia. :(');
+      console.log("Falhou.");
+      console.log(e);
+    }
+  }
+};
+//Lançar erro nas funções send e tratar na main
+async function sendImageExplanation(ctx, json) {
+  if (json.data[0]) {
+    try {
+      await ctx.reply(json.data[0].explanation);
+      console.log("Random explanation sent.");
+    } catch (e) {
+      await ctx.reply('Falha ao recuperar a imagem do dia. :(');
+      console.log("Falhou.");
+      console.log(e);
+    }
+  } else {
+    try {
+      await ctx.reply(json.data.explanation);
+      console.log("Explanation sent.");
+    } catch (e) {
+      await ctx.reply('Falha ao recuperar a imagem do dia. :(');
+      console.log("Falhou.");
+      console.log(e);
+    }
+  }
+};
 
 (async () => {
-    let jsonNASA = await getJSONAPOD(URL);
+
+    let jsonNASA;
+    try {
+      jsonNASA = await getJSONAPOD(URL);
+    } catch (e) {
+      jsonNASA = null;
+    };
     //Atualizar JSON a cada hora
-    setInterval(async function(){ jsonNASA = await getJSONAPOD(URL); }, 3600*10^3);
+    setInterval(async function(){
+      try {
+        jsonNASA = await getJSONAPOD(URL);
+      } catch (e) {
+        jsonNASA = null;;
+    };}, ONE_HOUR);
```

Comment:
Eu to com um pouco de dificuldade pra entender o que está acontecendo aqui. 

1. Precisa do primeiro try-catch se vc já faz a mesma coisa imediatamente após entrar no `setInterval`?  
2. A ideia desse procedimento é manter o `jsonNASA` sempre atualizado pra quando alguém enviar um comando o bot responder com a informação mais atual ?

Reply:
1. Não precisa. Eu não havia lido a ref do `setInterval`. Li agora e entendo que não precisa.
2. A ideia é essa, sim. Não quis colocar dentro do comando, da maneira que fiz em /random porque não existe a necessidade de verificar com frequência. Pretendo colocar um `setTimeOut` e executar uma vez por dia.

Reply:
De fato `setInterval` não chama a função imediatamente.
Tem que chamar a primeira vez.